common/threadable.h

```diff
@@ -28,32 +29,39 @@ class Threadable
     //! Run on the current thread, blocking until process ends
     virtual void run() = 0;
 
+    //! Check if the run() function has been called
+    virtual bool isRunning()
```

Comment:
Would it maybe be better to make the virtual function protected and make ``run()`` set the flag and call the internal virtual? The current approach seems a little error-prone as I can easily imagine forgetting to call ``runStart()`` if I was writing a Threadable.

Reply:
That's a much better idea!

Reply:
Heh. In fixing this up I already found somewhere where I forgot to call ``runStart()`` :-p